certora/specs/SocialRecoveryModule.spec

```diff
@@ -502,3 +503,113 @@ rule finalizeRecoveryAlwaysPossible(env e) {
 
     assert !isReverted, "legitimate recovery finalization reverted";
 }
+
+// This rule verifies that if recovery request data is changed, it must be one of the following functions:
+// - confirmRecovery(...)
+// - multiConfirmRecovery(...)
+// - executeRecovery(...)
+// - finalizeRecovery(...)
+// - cancelRecovery(...)
+// Each of these either updates or deletes the recovery request.
+rule recoveryRequestsChange(env e, calldataarg args, method f) {
```

Comment:
uber-style-nit: I would move `env` and `calldataarg` to the function body as it feels slightly unrelated to the rule parameterization (where you are saying "which function can do what" and the env and data are more of a small detail).

Reply:
Done 👍🏾